connectors/es/sink.py

```diff
@@ -339,6 +340,8 @@ async def get_docs(self, generator):
         Extraction happens in a separate task, when a document contains files.
         """
         logger.info("Starting doc lookups")
+        generator = TracedAsyncGenerator(generator, "get a doc")
```

Comment:
Why do you need this generator here?

Reply:
it's used to wrap the generator that fetches documents (get_docs) so we can trace how long each call to that generator takes

Reply:
Can you use the same format you use to decorate `_batch_bulk`?

Reply:
Ah good point, I can detect it's an async generator, I have not thought about this, thanks

Reply:
Done. That was fun!